rust-runtime/aws-smithy-types/src/big_number.rs

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! Big number types represented as strings.
+//!
+//! These types are simple string wrappers that allow users to parse and format
+//! big numbers using their preferred library.
+
+/// A BigInteger represented as a string.
+///
+/// This type does not perform arithmetic operations. Users should parse the string
+/// with their preferred big integer library.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct BigInteger(String);
+
+impl BigInteger {
+    /// Creates a new `BigInteger` from a string.
+    pub fn new(value: impl Into<String>) -> Self {
+        Self(value.into())
+    }
+
+    /// Returns the string representation.
+    pub fn as_str(&self) -> &str {
+        &self.0
+    }
```

Comment:
we shouldn't expose methods like this since they will probably be impossible to implement if we eventually switch to using a real internal representation that's not a string